### PR TITLE
Api source

### PR DESCRIPTION
**Pull request by felipe74590**
Changed API source to Predicthq API.

### CODE REVIEW

File: events_api.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import json
+from datetime import datetime
+
+import requests
+from decouple import config
+
+from web_scraper import Event
+
+base_url = "https://api.predicthq.com/v1/events/"
+access_token = config("EVENTS_HQ_TOKEN")
```
Comment (bbelderbos) [inline events_api.py:10]: does this expire? usually access tokens are obtained through the api ...

File: events_api.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+headers = {"Authorization": "Bearer " + access_token, "Accept": "application/json"}
+params = {"q": "Austin,Texas"}
+
+response = requests.get(url=base_url, headers=headers, params=params)
```
Comment (bbelderbos) [inline events_api.py]: quick test right? please put in a function

File: events_api.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+austin_data = json.loads(response.text)
```
Comment (bbelderbos) [inline events_api.py]: no json?

File: events_api.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+
+for atx_event in austin_data["results"]:
+    # COULD_DO: should I use google location api to turn address into venue names
```
Comment (bbelderbos) [inline events_api.py]: could do, does this just give the coordinates, ones you have/store those you can easily use google/openstreets indeed

File: events_api.py (hunk 3 above)
Comment (felipe74590) [reply events_api.py]: No it returns as text, so json.loads() is what allows me to access the dict values.

File: events_api.py (hunk 4 above)
Comment (felipe74590) [reply events_api.py]: This give coordinates and normal street addresss

File: events_api.py (hunk 1 above)
Comment (felipe74590) [reply events_api.py:10]: No the documentation does not say it expires, from by experience and according to the documentation Authorization Bearer tokens are usually set up this way, but obtained through a lockbox or secrets manager during runtime.

File: events_api.py (hunk 2 above)
Comment (felipe74590) [reply events_api.py]: Yes, I will put this in a function.

File: events_api.py (hunk 3 above)
Comment (bbelderbos) [reply events_api.py]: that's odd they don't return json, can I just sign up for free to get a token to try?

File: events_api.py (hunk 1 above)
Comment (bbelderbos) [reply events_api.py:10]: fair enough, I was too much into my own django rest thing lol, getting a key per account is common as well, e.g. openweather api

File: events_api.py (hunk 3 above)
Comment (felipe74590) [reply events_api.py]: Yes I think there's a two week trial

File: events_api.py [hunk 5]
```diff
@@ -4,28 +4,60 @@
 import requests
 from decouple import config
 
-from db_helper import add_events_to_db
 from web_scraper import Event
 
-base_url = "https://api.predicthq.com/v1/events/"
+base_url = config("PREDICT_API")
 access_token = config("EVENTS_HQ_TOKEN")
+LIMIT = 50 #max events per page
```
Comment (bbelderbos) [inline events_api.py:11]: you could add the comment info in the variable: `MAX_EVENTS_PER_PAGE`

File: main.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import argparse
```
Comment (bbelderbos) [inline main.py:1]: good move, you could also name this file `__main__.py` to make it the entry point to your package, Pybites tip:

image

Comment (bbelderbos) [reply main.py:1]: But for that you first would need to organize the code into a package folder, we can do this later ...

**Pull request merged**